### PR TITLE
[React Components] Refresh the page after Sign In / Sign Out

### PR DESCRIPTION
**Pull request by alkim91**
Refresh the page and redirect users back to the main storefront once they sign out.

## Description

I added the baseUrl(Storefront URL) in the App configs, and in the event handler for signing out, we redirect the user to the main storefront once they click 'Sign Out'. The baseUrl is configured based on the main storefront url(Storefront can be any language/country, not always US and English).

## Motivation and Context

If users are in any admin pages(Address book, account management), once they sign out we want them to be redirected back to the main storefront. Currently, we revoke the user token and reset cookies as well as the contexts, but no refresh/redirect takes place.

## How Has This Been Tested?

This has been tested in the two sign out user cases.

1. Signing out from the main navigation. Once we click 'sign out', no matter which page we're in(Catalog, address book, etc), we're taken back to the storefront.

2. Same thing happens when signing out from the account panel in responsive mode.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/main/components/App/config.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ export default {
         minicart: '#minicart'
     },
     pagePaths: {
-        addressBook: '/content/venia/us/en/my-account/address-book.html'
+        addressBook: '/content/venia/us/en/my-account/address-book.html',
+        baseUrl: document.querySelector('.header__logo').parentElement.pathname
```
Comment (dplaton) [inline ui.frontend/src/main/components/App/config.js:26]: I'm not sure if this is a reliable way to detect the home-page of the store.

**Review by dplaton — APPROVED** [review]

File: ui.frontend/src/main/components/App/config.js (hunk 1 above)
Comment (mhaack) [reply ui.frontend/src/main/components/App/config.js:26]: Actually the path should also not be here. Assume someone us using it with its own site and use a different site name or  change the `my-account` page name.

Comment (alkim91) [reply ui.frontend/src/main/components/App/config.js:26]: Got it. What's the best way to detect the homepage of the store, and where should I set this?

Comment (mhaack) [reply ui.frontend/src/main/components/App/config.js:26]: I don't know for JS, unfortunately. It is not that simple I think. In Java there are Util classes for to get site root path or URL, which will also consider URL rewrites etc.

Comment (alkim91) [reply ui.frontend/src/main/components/App/config.js:26]: @herzog31  thoughts? initially i thought if we can safely assume the logo is a constant element that redirects to the storefront, i could use it. but it looks like that assumptions not safe anymore like Markus said

Comment (herzog31) [reply ui.frontend/src/main/components/App/config.js:26]: Just an idea, we could do it this way:
* Get the base url like in the header component: https://github.com/adobe/aem-core-cif-components/blob/master/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/header/HeaderImpl.java#L55-L66
* Add it to the `StoreViewExporter`: https://github.com/adobe/aem-core-cif-components/blob/master/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/storeconfigexporter/StoreConfigExporter.java
* Expose it via the page: https://github.com/adobe/aem-core-cif-components/blob/master/ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/page.html#L24
* Consume via JavaScript.

@dplaton @mhaack @alkim91 wdyt?

Comment (mhaack) [reply ui.frontend/src/main/components/App/config.js:26]: Yes makes sense to me. That way we keep it consistent via the `StoreConfigExporter` all config related attributes in one place.

Comment (dplaton) [reply ui.frontend/src/main/components/App/config.js:26]: So we'd just configure all the paths in AEM? I'm ok with that, although it requires a different JIRA issue for refactoring the Address Book path as well.

Comment (herzog31) [reply ui.frontend/src/main/components/App/config.js:26]: @dplaton We have those already at https://github.com/adobe/aem-core-cif-components/blob/master/ui.apps/src/main/content/jcr_root/apps/core/cif/components/structure/page/v1/page/_cq_dialog/.content.xml#L76-L89. We just don't expose them _yet_ :)

Comment (dplaton) [reply ui.frontend/src/main/components/App/config.js:26]: Oh, we have _some_ of those :)
Ok, let's expose them.

Comment (alkim91) [reply ui.frontend/src/main/components/App/config.js:26]: Thanks for the suggestions! I'll re-implement today 👍

**Comment by dplaton** [comment]
I think this PR should be filed against the `dev` branch...

**Pull request closed without merging**